### ISSUE TITLE
MME misses some attaches/detaches  on multiple UE test (gxgy04.nlt)

### ISSUE THREAD

**Issue by uri200** [magma/magma#3625]
[https://app.zenhub.com/files/170803235/9c25db93-4a86-4e5f-a05c-91399413b419/download](https://app.zenhub.com/files/170803235/9c25db93-4a86-4e5f-a05c-91399413b419/download)Test04 runs several UEs in parallel attaching ,sending data and waiting for all of them to detach. Then it tries again the same. It repeats that 3 times.

When we run this test some attaches and detaches seems to be missing. We should have the same number of attaches and detaches but per the picture below, we only het 32 detaches.

I include the mme log while running this test 

![image.png](https://images.zenhubusercontent.com/5f84991a311cec52aab4eeed/a2ee0ea3-3941-4edb-b4b1-856dab3f8943)
[mme.log 2.zip](https://github.com/magma/magma/files/5497631/mme.log.2.zip)
[gx_gy_combined_04_201105_134207-1-1.pcapng.zip](https://github.com/magma/magma/files/5497717/gx_gy_combined_04_201105_134207-1-1.pcapng.zip)

**Comment by uri200** [comment]
more logs
96 UEs attach, but only 64 detach request succeeded

[detach_logs.zip](https://github.com/magma/magma/files/5569340/detach_logs.zip)

**Comment by ulaskozat** [comment]
@uri200 checked out the pcap for the last file. Indeed there are 96 attaches and 64 detaches. But, I do not see the detach request coming in from the UEs in the last round in this pcap (3rd round). Looks like a TeraVM issue.

**Comment by ulaskozat** [comment]
in the previous pcap S1 connection is being reinitialized in the middle of the test. do we have syslog data for these?

**Comment by uri200** [comment]
Hey @ulaskozat it looked like the core dumps were from previous runs when stateless was enabled. I removed them and ran the test for few days and no core dumps were generated.


Ulrich took a look and found again an issue with repeated TMSI (even that we added the random behaviour): On the attached pcap (also showing in the picture), we can see the same TMSI (271360) being used by two different IMSIs.

The behaviour we added with TMIS was a to use a pseudorandom number. That means there will be no reuse of a TIMSI. The number given by Magma on the `Initial Context Setup Request` should be always different. The pseudo random sequence is big enough not to repeat again for long time. But here we see that for some reason MME is giving the same TMSI again to a different UE 30 s later.

In fact, if you filter by TMSI 	using this filter `nas_eps.emm.m_tmsi == 271360` you will see that number is used in 4 different Initial Context Setup Requests


Filter used on wireshark
`s1ap.MME_UE_S1AP_ID == 418 or s1ap.MME_UE_S1AP_ID == 416`
![image](https://user-images.githubusercontent.com/16157139/100659036-de749480-3304-11eb-8965-f1cc8839c070.png)

[gx_gy_combined_04_201128_033835-1-1.pcapng.zip](https://github.com/magma/magma/files/5618464/gx_gy_combined_04_201128_033835-1-1.pcapng.zip)
